src/main/java/Duke.java

```diff
@@ -1,10 +1,190 @@
+
+            //if the item is to be marked or unmarked, follow the correct steps to extract the index
+            if (matchesMark || matchesUnmark) {
+                markOrUnmark(matchesMark, input);
+
+                //Otherwise, create a new task
+            } else {
+                Task task = createTask(input);
+                if (task == null) {
+                    System.out.println(H_LINE + INDENT
+                            + "Looks like there was an issue with adding this task: "
+                            + input + INDENT + "Please try again!" + H_LINE + "\n");
+                }
+                System.out.println(H_LINE + INDENT
+                        + "Success!" + INDENT + "Added: " + (listIndex + 1) + ". " + task + H_LINE + "\n");
+                taskList[listIndex] = task;
+                listIndex++;
+            }
+        }
+        return shouldExit;
+    }
+
+    private static Task createTask(String input) {
+        Task task;
+        if (input.startsWith("deadline")) {
+            task = createDeadline(input);
+            if (task == null) {
+                System.out.println(INDENT
+                        + "Was unable to create a Deadline! Please specify a date after writing /by.");
+            }
+        } else if (input.startsWith("event")) {
+            task = createEvent(input);
+            if (task == null) {
+                System.out.println(INDENT
+                        + "Was unable to create an Event! Please specify a date after writing /at.");
+            }
+        } else {
+            task = new Task(input);
+        }
+        return task;
+    }
+
+    /* Create a new Deadline with the correct due date */
+    private static Task createDeadline(String input) {
+        int dateIndex = input.indexOf("/by");
+        if (dateIndex == -1) {
+            return null;
+        }
+        int startIndex = findNextLetter("deadline", input);
+        int endIndex = dateIndex + findNextLetter("/by", input.substring(dateIndex));
+        return new Deadline(input.substring(startIndex, dateIndex), input.substring(endIndex));
+    }
+
+    /* Create a new Event with the correct date */
+    private static Task createEvent(String input) {
+        int dateIndex = input.indexOf("/at");
+        if (dateIndex == -1) {
+            return null;
+        }
+        int startIndex = findNextLetter("event", input);
+        int endIndex = dateIndex + findNextLetter("/at", input.substring(dateIndex));
+        return new Event(input.substring(startIndex, dateIndex), input.substring(endIndex));
+    }
+
+    private static int findNextLetter(String word, String input) {
+        int index = word.length();
+        while (input.charAt(index) == ' ') {
+            index++;
+        }
+        return index;
+    }
+
+    /* Either mark or unmark a task */
+    private static void markOrUnmark(boolean toMark, String input) {
+        String type = toMark ? "Mark" : "Unmark";
```

Comment:
Can use better naming here for the variable to bring meaning to it instead of merely naming it as type.